### PR TITLE
feat: additional watchevent types for metrics

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

This adds `WatchEvent` types for consuming metrics from the informer.

Please look at [Watch Config](https://github.com/defenseunicorns/kubernetes-fluent-client/blob/8f6bed408fc967fb4f68b60001cf8a8dc5f7bc5e/src/fluent/watch.ts#L49), as some configuration options have been renamed.

## Related Issue

Fixes #https://github.com/defenseunicorns/pepr/pull/983

Relates to # https://github.com/defenseunicorns/pepr/issues/977

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

BREAKING CHANGE:  This changes the names on the WatchConfig. Look at the WatchConfig as some configuration options have been renamed.

### CODE REVIEW

File: src/fluent/watch.ts [hunk 1]
```diff
@@ -34,12 +34,18 @@ export enum WatchEvent {
   LIST = "list",
   /** List operation error */
   LIST_ERROR = "list_error",
+  /** Cache Misses */
+  CACHE_MISS = "cache_miss",
+  //** Increment resync failure count */
```
Comment (cmwylie19) [inline src/fluent/watch.ts:39]:
```suggestion
  /** Increment resync failure count */
```

File: src/fluent/watch.ts [hunk 2]
```diff
@@ -34,12 +34,18 @@ export enum WatchEvent {
...
+  INIT_CACHE_MISS = "init_cache_miss",
 }
 
 /** Configuration for the watch function. */
 export type WatchCfg = {
   /** The maximum number of times to retry the watch, the retry count is reset on success. Unlimited retries if not specified. */
-  retryMax?: number;
+  resyncFailureMax?: number;
   /** Seconds between each retry check. Defaults to 5. */
   retryDelaySec?: number;
```
[the remark this answers is not in the document]
Comment (cmwylie19) [reply src/fluent/watch.ts:50]: Would you prefer `window` since that is the label in the gauge?

[the remark this answers is not in the document]
Comment (cmwylie19) [reply src/fluent/watch.ts:50]: oops disregard, this is the wrong one

[the remark this answers is not in the document]
Comment (cmwylie19) [reply src/fluent/watch.ts:50]: yeah i see, that does make sense

**Review by btlghrants — APPROVED** [review]
After talking it through synchronously, LGTM.

**Pull request merged**